x-pack/plugins/alerting/server/rules_client/common/retry_if_bulk_edit_conflicts.ts

```diff
@@ -76,12 +80,17 @@ export const retryIfBulkEditConflicts = async (
     const results = [...accResults, ...resultSavedObjects.filter((res) => res.error === undefined)];
     const apiKeysToInvalidate = [...accApiKeysToInvalidate, ...localApiKeysToInvalidate];
     const errors = [...accErrors, ...localErrors];
+    // Create array of unique skipped rules by id
+    const skipped = [
+      ...new Map([...accSkipped, ...localSkipped].map((item) => [item.id, item])).values(),
+    ];
```

Comment:
Why do you need to do this operation to guarantee that all skipped rules have unique id?
As I see, they will be already unique.

Reply:
This is to make sure that we prevent skipped rules from potentially accumulate the same skip results multiple times when this function is be called recursively. That's the test case I added in `x-pack/plugins/alerting/server/rules_client/common/retry_if_bulk_edit_conflicts.test.ts` which had the issue with the casting.

Reply:
Based on code `skipped` rules will never be added to `rules` which will be pass to `result = await context.unsecuredSavedObjectsClient.bulkCreate(rules, { overwrite: true });`.